install/management/appliance.go

```diff
@@ -0,0 +1,395 @@
+	"github.com/vmware/vic/pkg/vsphere/tasks"
+	"github.com/vmware/vic/pkg/vsphere/vm"
+
+	"golang.org/x/net/context"
+)
+
+var (
+	lastSeenProgressMessage string
+	unitNumber              int32
+)
+
+func (d *Dispatcher) isVCH(vm *vm.VirtualMachine) (bool, error) {
+	if vm == nil {
+		return false, errors.New("nil parameter")
+	}
+
+	info, err := vm.FetchGuestInfo(d.ctx)
+	if err != nil {
+		err = errors.Errorf("Failed to fetch guest info of appliance vm, %s", err)
+		return false, err
+	}
+	if _, ok := info["guestinfo.vch/components"]; ok {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (d *Dispatcher) removeApplianceIfForced(conf *configuration.Configuration) error {
+	vm, err := d.findAppliance(conf)
+	if err != nil {
+		return err
+	}
+	log.Debugf("Appliance is found")
+	if vm != nil && d.force {
+		if ok, err := d.isVCH(vm); !ok {
+			err = errors.Errorf("VM %s is found, but is not VCH appliance, please choose different name", conf.DisplayName)
+			return err
+		}
+		log.Infof("Appliance exists, remove it...")
+		_, err = tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+			return vm.PowerOff(ctx)
+		})
+		if err != nil {
+			err = errors.Errorf("Failed to power off existing appliance: %s", err)
+			return err
+		}
+		_, err = tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+			return vm.Destroy(ctx)
+		})
+		if err != nil {
+			err = errors.Errorf("Failed to destroy existing appliance: %s", err)
+			return err
+		}
+		m := object.NewFileManager(d.session.Client.Client)
+		if _, err = tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+			return m.DeleteDatastoreFile(ctx, d.session.Datastore.Path(conf.DisplayName), d.session.Datacenter)
+		}); err != nil {
+			err = errors.Errorf("Failed to remove existing VCH data files, %s", err)
+			return err
+		}
+	} else if vm != nil {
+		err = errors.Errorf("VM already exists with display name %s. Name must be unique. Exiting...", conf.DisplayName)
+		return err
+	}
+
+	return nil
+}
+
+func (d *Dispatcher) getNetworkDevices(conf *configuration.Configuration) ([]types.BaseVirtualDeviceConfigSpec, error) {
+	var devices []types.BaseVirtualDeviceConfigSpec
+	var err error
+
+	// bridge network
+	network, err := d.session.Finder.NetworkOrDefault(d.ctx, conf.BridgeNetworkPath)
+	if err != nil {
+		err = errors.Errorf("Failed to get bridge network: %s", err)
+		return nil, err
+	}
+	d.networks["bridge"] = network
+	d.nics[conf.BridgeNetworkName] = "bridge"
+
+	// client network
+	network, err = d.session.Finder.NetworkOrDefault(d.ctx, conf.ExtenalNetworkPath)
+	if err != nil {
+		err = errors.Errorf("Failed to get external network: %s", err)
+		return nil, err
+	}
+	d.networks["client"] = network
+	d.nics[conf.ExtenalNetworkName] = "client"
+
+	// management network
+	if conf.ManagementNetworkName != "" {
+		network, err = d.session.Finder.NetworkOrDefault(d.ctx, conf.ManagementNetworkPath)
```

Comment:
This should not be NetworkOrDefault - if a name is specified and it cannot be found it's an error.